app/src/hooks/activity-value-form/use-activity-form.ts

```diff
@@ -0,0 +1,78 @@
+import { useForm } from "react-hook-form";
+import { useEffect } from "react";
+import { ActivityValue } from "@/models/ActivityValue";
+import { SuggestedActivity } from "@/util/form-schema";
+import { Inputs } from "@/components/Modals/activity-modal/activity-modal-body";
+
+const useActivityForm = ({
+  targetActivityValue,
+  selectedActivity,
+}: {
+  targetActivityValue: ActivityValue | undefined;
+  selectedActivity?: SuggestedActivity;
+}) => {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    watch,
+    setError,
+    setFocus,
+    formState: { errors },
+  } = useForm<Inputs>();
+
+  useEffect(() => {
+    if (targetActivityValue) {
+      reset({
+        activity: {
+          ...targetActivityValue.activityData,
+          dataQuality: targetActivityValue?.dataSource?.dataQuality,
+          sourceReference: targetActivityValue?.dataSource?.notes,
+          CH4EmissionFactor: targetActivityValue?.activityData?.ch4_amount,
+          CO2EmissionFactor: targetActivityValue?.activityData?.co2_amount,
+          N2OEmissionFactor: targetActivityValue?.activityData?.n2o_amount,
+          activityDataAmount: 0,
+          activityDataUnit: null,
+          emissionFactorType: "",
```

Comment:
This needs to be loaded from the ActivityValue/ GasValue as well. The type for the drop down in this case is the `name` of the data source associated with the EmissionsFactor (as you can see in how the data for the dropdown is created).

All of the other fields need to be loaded here as well, not just the ones above.

Reply:
So these values are currently not being stored in the DB, I've opted to place them in the metadata column for now